Makefile

```diff
@@ -64,7 +63,15 @@ yarn-install:
 
 .PHONY: serve
 serve: yarn-install
-	$(DOCKER_CMD) yarn serve
+	docker run \
```

Comment:
I'll defer to @krancour who I think set this Makefile up last, but I'm wondering if we can make tweaks to the main `DOCKER_CMD` so that this target can be aligned w/ others, e.g. `$(DOCKER_CMD) yarn serve` (or similar).

A couple potential ways:

- Always add `-p 4200:4200` to `DOCKER_CMD` the definition?
- Split `DOCKER_CMD` up into `DOCKER_CMD_PREFIX` sans `DEV_IMAGE` such that `DOCKER_CMD := $(DOCKER_CMD_PREFIX) $(DEV_IMAGE)` and this target can become `$(DOCKER_CMD_PREFIX) -p 4200:4200 $(DEV_IMAGE) yarn serve`?
- Other?  (if worth it?)

Reply:
np—the `serve` target is the only target that needs docker port forwarding enabled (as far as I could tell), so I made that one the outlier so it's possible to run multiple targets at the same time, e.g. you can now run `shell` while running `test` while running `e2e`. Happy to go any which way y'all like best!

Reply:
> Split DOCKER_CMD up into DOCKER_CMD_PREFIX sans DEV_IMAGE such that DOCKER_CMD := $(DOCKER_CMD_PREFIX) $(DEV_IMAGE) and this target can become $(DOCKER_CMD_PREFIX) -p 4200:4200 $(DEV_IMAGE) yarn serve?

That would introduce a new complication. As it stands now, `$(DOCKER_CMD)` evaluates to the empty string if the `SKIP_DOCKER` env var isn't `true`. So the "magic" of `$(DOCKER_CMD)` is that if you don't want Docker (or are _already_ inside a container for whatever reason) all the `docker run ...` stuff is skipped and you get normal execution of the command that follows within your current environment. If we make the target look like this:

```
$(DOCKER_CMD_PREFIX) -p 4200:4200 $(DEV_IMAGE) yarn serve
```

Then when `SKIP_DOCKER` is `true`, the command that will execute is `-p 4200:4200 yarn serve`, which is obviously problematic.

So... I think I remain in favor of the first option, which is how it was originally:

> Always add -p 4200:4200 to DOCKER_CMD the definition?

Sure, that port will never be used for most cases, but I think it's good enough.

Everything else in this PR, however, is 👌 .

Good to see you @seanknox and thanks!

Reply:
> so I made that one the outlier so it's possible to run multiple targets at the same time, e.g. you can now run shell while running test while running e2e

Although this is a good point that I overlooked.

Let's merge this as is and I'll figure something out so my concerns _and_ the above are both addressed.

LGTM.